docs/proposals/20210316-containerrecreaterequest.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Proposal Template
+authors:
+  - "@FillZpp"
+reviewers:
+  - "@Fei-Guo"
+  - "@furykerry"
+creation-date: 2021-03-16
+last-updated: 2021-03-16
+status: implementable
+---
+
+# ContainerRecreateRequest
+
+- Provide a way to let users restart/recreate one or more containers in a Pod.
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a proposal and for highlighting
+any additional information provided beyond the standard proposal template.
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc) a table of contents from markdown are available.
+
+   * [ContainerRecreateRequest](#containerrecreaterequest)
+      * [Table of Contents](#table-of-contents)
+      * [Motivation](#motivation)
+      * [Proposal](#proposal)
+         * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Implementation History](#implementation-history)
+
+## Motivation
+
+Now that the only way to recreate a container in an active Pod is to update it in-place.
+However, it also requires users to change the image defined in workloads (e.g. CloneSet/Advanced StatefulSet).
+
+We hope to provide a way to let users recreate(restart) one or more containers in any active Pods.
+It is useful for some user scenarios, such as:
+
+1. Restart the application in app container.
+2. Restart a sidecar contaienr along with app container.
+3. Force the init container running once again.
```

Comment:
please identify the side-effect of rerun the init container, and maybe kruise should not cover such feature for now

Reply:
removed temporarily